src/compiler/checker.ts

```diff
@@ -9613,6 +9614,9 @@ namespace ts {
                             context.inferredTypes[index] = inferredType = instantiatedConstraint;
                         }
                     }
+                    if (context.useAnyForNoInferences && !inferences.length && inferredType === emptyObjectType) {

```

Comment:
Ah, I thought the new code was in a different place but I see now. Anyway, one issue is that you'll change *any* `emptyObjectType` into an `any` regardless of how that `emptyObjectType` came about. It might in fact have been produced by an actual inference.